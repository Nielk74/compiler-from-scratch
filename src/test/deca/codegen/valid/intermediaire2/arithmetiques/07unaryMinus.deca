// Description:
//    Programme testant le moins unaire (avec et sans op√©rations binaires)
//
// Resultats:
//    07unaryMinus.expected
//
// Historique:
//    cree le 10/06/2022

{
    int a = 5;
    float b = 7.5;
    float c;
    int d = 3;

    println(-a);
    a = -a;
    println(a);

    println(-b);
    b = -b;
    println(b);

    println(-(a + b));
    c = -(a + b);
    println(c);

    println(-(a - b));
    c = -(a - b);
    println(c);

    println(-(a * b));
    c = -(a * b);
    println(c);

    println(-(a / b));
    c = -(a / b);
    println(c);

    println(-(a % d));
    d = -(a % d);
    println(d);

    println(-(-(a + b)));
    c = -(-(a + b));
    println(c);
    
    // conv float avec moins unaire
    c = -d;
    println(c);
}