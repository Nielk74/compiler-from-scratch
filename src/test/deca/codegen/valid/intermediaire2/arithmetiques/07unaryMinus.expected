-5
-5
-7.50000e+00
-7.50000e+00
1.25000e+01
1.25000e+01
-2.50000e+00
-2.50000e+00
-3.75000e+01
-3.75000e+01
-6.66667e-01
-6.66667e-01
2
2
-1.25000e+01
-1.25000e+01
-2.00000e+00
